test/unit/Services/HttpServiceTest.php

```diff
@@ -169,6 +169,38 @@ static function ($headers) use ($locale) {
         $httpServiceMock->send('/my/uri/123', $resource);
     }
 
+    /**
+     * Verify 'CLIENTIP' header only set when a clientIp is defined in the Unzer object.
+     *
+     * @test
+     * @dataProvider clientIPHeaderShouldOnlyBeSetIfPropertyIsNotEmptyDP
+     *
+     * @param $clientIp
+     */
+    public function clientIPHeaderShouldOnlyBeSetIfPropertyIsNotEmpty($clientIp): void
+    {
+        $httpServiceMock = $this->getMockBuilder(HttpService::class)->setMethods(['getAdapter'])->getMock();
+        $adapterMock = $this->getMockBuilder(CurlAdapter::class)->setMethods(['setHeaders', 'execute'])->getMock();
+        $httpServiceMock->method('getAdapter')->willReturn($adapterMock);
+
+        $unzer = new Unzer('s-priv-MyTestKey', $clientIp);
+        $unzer->setClientIp($clientIp);
+        $resource = (new DummyResource())->setParentResource($unzer);
+
+        /** @noinspection PhpParamsInspection */
+        $adapterMock->expects($this->once())->method('setHeaders')->with(
+            $this->callback(
+                static function ($headers) use ($clientIp) {
+                    return $clientIp === ($headers['CLIENTIP'] ?? null);
```

Comment:
This assertion is not enough since it does not explicitly make sure the header is not there when no IP has been set.

Reply:
I changed the aproach here, focusing on the header composing method itself. We can discuss that later.

Reply:
updated test according to our discussion. Please check again.